lib/matplotlib/axes/_base.py

```diff
@@ -2185,6 +2186,10 @@ def grid(self, b=None, which='major', axis='both', **kwargs):
             b = True
         b = _string_to_bool(b)
 
+        # We need to disable both grids before replotting the wanted ones
+        self.xaxis.grid(False)
```

Comment:
There was a bug here.
Example : 
```python
plt.grid(axis='both')
plt.grid(axis='x')
```
You end up here with the grid on the 'y' axis, instead of 'x'.

Example 2 : 
``` python
plt.grid(True, axis='both')
plt.grid(True, axis='x')
```
You end up here with both grids, instead of only the x grid.

Removing the grid on each axis before plotting fixes this issue.

I was not able to make a parameter for choosing which grid to plot at first because of this issue